pipelines/jobs/pipeline_job_template.groovy

```diff
@@ -94,7 +94,7 @@ pipelineJob("${BUILD_FOLDER}/${JOB_NAME}") {
         stringParam('dockerExcludes', '', 'Map of targetConfigurations to exclude from docker building. If a targetConfiguration (i.e. { "x64LinuxXL": [ "openj9" ], "aarch64Linux": [ "hotspot", "openj9" ] }) has been entered into this field, jenkins will build the jdk without using docker. This param overrides the dockerImage and dockerFile downstream job parameters.')
         stringParam('baseFilePath', '', "Relative path to where the build_base_file.groovy file is located. This runs the downstream job setup and configuration retrieval services.<br>Default: <code>${defaultsJson['baseFileDirectories']['upstream']}</code>")
         stringParam('buildConfigFilePath', '', "Relative path to where the jdkxx_pipeline_config.groovy file is located. It contains the build configurations for each platform, architecture and variant.<br>Default: <code>${defaultsJson['configDirectories']['build']}/jdkxx_pipeline_config.groovy</code>")
-        choiceParam('releaseType', [jobReleaseType, 'Nightly', 'Nightly Without Publish', 'Weekly', 'Release'], 'Nightly - release a standard nightly build.<br/>Nightly Without Publish - run a nightly but do not publish.<br/>Weekly - release a standard weekly build, run with extended tests.<br/>Release - this is a release, this will need to be manually promoted.')
+        choiceParam('releaseType', [jobReleaseType, 'Nightly', 'Nightly Without Publish', 'Weekly', 'Release'].unique(), 'Nightly - release a standard nightly build.<br/>Nightly Without Publish - run a nightly but do not publish.<br/>Weekly - release a standard weekly build, run with extended tests.<br/>Release - this is a release, this will need to be manually promoted.')
```

Comment:
does it even support function to be called on this level ?
or if  you have a screenshot after your change

Reply:
i would say to call unique() is better done in https://github.com/adoptium/ci-jenkins-pipelines/blob/22a9cf2a46d787bcf5fa27d5f88317974f5fdb9f/pipelines/jobs/pipeline_job_template.groovy#L13 after convert jobReleaseType to list jobReleaseTypeList by adding the rest into it

Reply:
Heya Wen,

I think you can use unique this way because you can see [here](https://riptutorial.com/groovy/example/18010/remove-duplicates) that unique works on a list of strings, and you can see [here](https://blog.mrhaki.com/2010/01/groovy-goodness-removing-duplicate.html) (line 17) that Groovy allows you to call methods directly on an array without assigning the array to a named variable.

I did try something similar to what you suggest, but this seemed simpler. Plus, it keeps all the data in one place, next to the choice parameter we're populating.